tests/data/src/required_by_mixed_capitalization/setup.py

```diff
@@ -0,0 +1,6 @@
+from setuptools import setup
+
+setup(name='requires_Capitalized',
```

Comment:
Sorry, in hindsight my previous suggestion was not very clear. I think that this package should be named "A" (or anything else, it doesn't matter), and then it should have `requires_Capitalized` in its `install_requires` list. That would ensure that the names of things in `install_requires` that we're comparing against are also normalized.

Reply:
I see. 
As you can see by the following commits, I messed it up a bit.
Anyway, do you prefer having both the existing test and the new one? Just the new one?

Reply:
No worries! I would keep `test_show_required_by_packages_capitalized`, remove `test_show_required_by_with_mixed_capitalization`, and add the new one (maybe called `test_show_required_by_requires_capitalized`).

Reply:
By the way, I tested it earlier and it seemed to me that install_requires isn't being normalized.
I'm rechecking now but anyway, shouldn't the fix and test be somewhere else?
It doesn't seem related here. 
I mean, I don't mind doing it here in this PR on the fly. but it seems out of place for me.

Reply:
I tried to do that earlier, but it didn't install the package because of:
`ERROR: No matching distribution found for requires_Capitalized==1.0 (from simple-1==1.0)`
(obviously)
So I figured I should Create a new tar.gz called Requires_Capitalized for this test.
I wanted to make sure with you first that this is in fact what you meant before I do so.

Reply:
I think it would be OK to explicitly install `Requires_Capitalized` before installing `A`, then we should not need another .tar.gz.

Reply:
Are you sure that installing it explicitly would work?
I've done the following:
(Don't mind the name, WIP and stuff)
1. Added a file
```
setup(name='Req_Cap_Todo',
      version='1.0',
      )
```


2. Installing this file:
```
 editable_path = os.path.join(data.src, 'Req_Cap_Todo')
    script.pip(
        'install', '--no-index', '-f', data.find_links, editable_path
    )
```
3. Installed the package
```
setup(name='simpleone',
      version='1.0',
      install_requires=['Req_Cap_Todo==1.0']
      )

```
And I get the result of 
`['Name: Req-Cap-Todo', 'Version: 1.0', ... ,  'Requires: ', 'Required-by: ']`

To make sure, I tried to installed Click first and black afterwards, but it did show the correct message (with my fix, of course)

Reply:
Installing explicitly was meant to avoid making another .tar.gz, and that looks like it worked. :+1:

The "Required-by" of "Req-Cap-Todo" missing "simpleone" is a similar bug to the original that we should also ensure works.

Reply:
Installing explicitly was meant to avoid making another .tar.gz, and that looks like it worked. :+1: The other item is a bug.